# Case Study

This is a case study with context-based questions and answers.

## Context
You are working at a retail company with both online and offline operations. Your manager hands you the sales dataset and says:

“We need a quick analysis. I want to identify underperforming regions, understand which product categories perform best in each region, and spot any unusual patterns in the data.”


## Basic Insights:

What are the total sales and profits overall?

Which region has the highest and lowest sales?

What is the average discount given?

In [66]:
import pandas as pd
df = pd.read_csv('case_study.csv')
df.head()

Order ID        Product     Category Region     Sales  Quantity  Discount  \
0      1001     Nike Shoes      Apparel  South  19288.00         2      0.00   
1      1002         iPhone  Electronics   West   2452.00         1      0.00   
2      1003  Action Figure         Toys  North  57506.75         5      0.15   
3      1004     Nike Shoes      Apparel  North  78630.10         2      0.15   
4      1005   Levi's Jeans      Apparel   East   5591.30         1      0.15   

     Profit Order Date  
0   4539.24  5/20/2023  
1    352.42  5/10/2023  
2   8286.04  5/31/2023  
3  13213.80   2/9/2023  
4    667.28  3/30/2023

In [67]:
df.shape

(150, 9)

In [68]:
df.isnull().sum()

Order ID      0
Product       0
Category      0
Region        0
Sales         0
Quantity      0
Discount      0
Profit        0
Order Date    0
dtype: int64

In [69]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 9 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Order ID    150 non-null    int64  
 1   Product     150 non-null    object 
 2   Category    150 non-null    object 
 3   Region      150 non-null    object 
 4   Sales       150 non-null    float64
 5   Quantity    150 non-null    int64  
 6   Discount    150 non-null    float64
 7   Profit      150 non-null    float64
 8   Order Date  150 non-null    object 
dtypes: float64(3), int64(2), object(4)
memory usage: 10.7+ KB


In [70]:
import datetime as dt
df['Order Date'] = pd.to_datetime(df['Order Date'])
df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 9 columns):
 #   Column      Non-Null Count  Dtype         
---  ------      --------------  -----         
 0   Order ID    150 non-null    int64         
 1   Product     150 non-null    object        
 2   Category    150 non-null    object        
 3   Region      150 non-null    object        
 4   Sales       150 non-null    float64       
 5   Quantity    150 non-null    int64         
 6   Discount    150 non-null    float64       
 7   Profit      150 non-null    float64       
 8   Order Date  150 non-null    datetime64[ns]
dtypes: datetime64[ns](1), float64(3), int64(2), object(3)
memory usage: 10.7+ KB


In [71]:
#What are the total sales and profits overall?
Total_sales = df['Sales'].sum().round(2)
Total_profits = df['Profit'].sum()
print(f'The Total sales are {Total_sales} and the Total Profits are Rs.{Total_profits}')

The Total sales are 13226095.15 and the Total Profits are Rs.2513891.5


In [72]:
#Which region has the highest and lowest sales?
df.groupby('Region')['Sales'].sum().sort_values(ascending=False)


Region
East     4868485.00
West     3414870.90
North    2650787.80
South    2291951.45
Name: Sales, dtype: float64

In [73]:
#What is the average discount given?
avg_dis = df['Discount'].mean()
avg_dis

0.08

## Product Insights:
Top 5 most sold products by quantity.

In [74]:
df.groupby('Product')['Quantity'].sum().sort_values(ascending= False).head(5)

Product
Dining Table    54
Nike Shoes      52
iPhone          48
Levi's Jeans    47
Lego Set        44
Name: Quantity, dtype: int64

In [75]:
#Top 3 product categories by sales.
df.groupby('Category')['Sales'].sum().sort_values(ascending= False).head(3)

Category
Electronics    3629980.35
Apparel        2604781.25
Furniture      2588036.55
Name: Sales, dtype: float64

## Regional Insights:
In each region, which product category performs best?

In [76]:
df.groupby(['Region','Category'])['Sales'].sum().reset_index().sort_values(by=['Region','Sales'], ascending=[True,False])

Region     Category       Sales
4    East         Toys  1175125.10
1    East  Electronics  1158450.00
3    East    Furniture   995776.00
2    East         Food   771835.20
0    East      Apparel   767298.70
6   North  Electronics   825624.05
5   North      Apparel   727472.15
8   North    Furniture   513882.45
9   North         Toys   406102.05
7   North         Food   177707.10
13  South    Furniture   578900.70
12  South         Food   563161.80
14  South         Toys   545820.40
10  South      Apparel   344828.70
11  South  Electronics   259239.85
16   West  Electronics  1386666.45
15   West      Apparel   765181.70
18   West    Furniture   499477.40
19   West         Toys   415045.30
17   West         Food   348500.05

Toys performs best in East Region,Electronics in North,Furniture in South, and Electronics again in West.

In [77]:
#Are there any regions where profits are consistently low?
df['month']= df['Order Date'].dt.month
df.groupby(['Region','month'])['Profit'].sum().sort_values(ascending = False)

Region  month
East    1        214398.18
West    5        187854.82
East    6        178089.28
        2        167098.47
North   3        145928.07
West    4        136726.84
South   2        135253.35
East    3        134490.34
North   1        128352.02
East    5        117318.19
        4        116233.75
South   3        113553.44
North   2         94933.94
West    1         90482.70
South   6         82607.00
North   4         73623.22
West    6         70651.86
        3         68221.49
South   1         57935.69
North   6         54257.39
West    2         49278.07
North   5         43097.98
South   5         32985.50
        4         20519.91
Name: Profit, dtype: float64

South Sales are lowest.

In [78]:
max(df['Order Date'])

Timestamp('2023-06-30 00:00:00')

In [79]:
min(df['Order Date'])

Timestamp('2023-01-01 00:00:00')

## Suspicious/Interesting Findings:
#Are there any discounts that lead to negative profits?

In [80]:
df['Profit'].corr(df['Discount'])

-0.038445392553860404

Profits and discount are weakly corelated. There are very less chances that discounts leads to negative profits.

## Build a very simple linear regression model to see if you can predict Profit using Sales, Discount, and Quantity.

In [81]:
import statsmodels.api as sm
X = df[['Discount','Sales','Quantity']]
y = df['Profit']

X = sm.add_constant(X)
model = sm.OLS(y,X).fit()

print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                 Profit   R-squared:                       0.794
Model:                            OLS   Adj. R-squared:                  0.790
Method:                 Least Squares   F-statistic:                     187.4
Date:                Sun, 27 Jul 2025   Prob (F-statistic):           7.44e-50
Time:                        21:25:01   Log-Likelihood:                -1530.7
No. Observations:                 150   AIC:                             3069.
Df Residuals:                     146   BIC:                             3082.
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const        953.7933   1475.656      0.646      0.5

This means 79.4% of the variance in Profit is explained by the three variables Discount, Sales, and Quantity.From the p values we can see that only Sales is statistically significant in predicting the profit (p<0.05) that means For every ₹1 increase in Sales, Profit increases by ~₹0.20. Overall the model is strong since Prob (F-statistic) = 7.44e-50 (<0.05) but there is multicollinearity in the data. We shall go ahead with VIF to assess it.